src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetGenerator.cs

```diff
@@ -365,23 +366,51 @@ public enum InstructionSet
         ILLEGAL = 0,
         NONE = 63,
 ");
+
             foreach (string architecture in _architectures)
             {
-                int counter = 1;
                 foreach (var jitName in _architectureJitNames[architecture])
                 {
-                    tr.WriteLine($"        {architecture}_{jitName}={counter++},");
+                    tr.WriteLine($"        {architecture}_{jitName} = InstructionSet_{architecture}.{jitName},");
                 }
             }
 
-            tr.Write(@"
-    }
+            tr.Write(@"    }
+
+");
+
+            foreach (string architecture in _architectures)
```

Comment:
It's probably easier to see these changes by looking at https://github.com/dotnet/runtime/pull/62420/files#diff-47b7c76a34a29a49750b1591fcf7ed6196296eb075ca71cc160b03ff5e717f49R21.

This basically just adds `InstructionSet_X64`, `InstructionSet_X86`, and `InstructionSet_ARM64` enums that contain only the ISAs for that architecture. These aren't actually used by anything, they are just returned by three new properties on `InstructionSetFlags` to make it simpler to debug.

Without this, you'll end up seeing the `ARM64_*` enum values regardless of platform due to how the debugger displays enums with overlapping values. This was making it practically impossible to debug since you'd see `ARM64_AdvSimd` rather than `X64_SSE` for example.